src/benchmark/benchmark_basic_fixture.cpp

```diff
@@ -22,10 +22,14 @@ void BenchmarkBasicFixture::SetUp(::benchmark::State& state) {
 
```

Comment:
> For test coverage:
> 
> * For `SimdBp128Packing` (and unpacking) only the bitcases 11-16 are covered.

Yes that’s true, but it’s just a tiny part of what is supposedly not covered.

> * `NsVector` methods like `NsVector::decode` are not used. I put prints in there and didn't see them while executing the tests

Yes that’s true, too. I added another tests for this method.

> * Same with `FixedSizeByteAlignedVector`, because none of the `_on_*` methods are called by the `NsVector`

This is not true. Have a look at `null_suppression_test.cpp`. The iterator methods and `create_base_decoder` are in fact called.

> * Same with `SimdBp128Iterator`, because `_on_cbegin` is never called

I tested this via print debugging and it gets in fact called.

> I can't really put my finger on why all this code is unused at the moment, but I'm sure you'll figure it out :)

Furthermore some of the new untested code, I didn’t even touch (`abstract_lqp_node.cpp`).

@mrks Any ideas how we can solve this?

Reply:
I just realized that you are 8 commits behind `master`, so you are probably missing my commits for all the LQP tests

Reply:
> Coverage — decreased by 4.3 compared to master and is now 85.7%

I took the liberty of merging the `master` but now it's even worse - oops.

> Yes that’s true, but it’s just a tiny part of what is supposedly not covered.

Actually, for `simd_bp128_packing.cpp` alone, the non-covered bitcases make up about 160 lines of code alone. I think that's pretty significant.

Other than that, after skimming the pycobertura report it seems like it's just a lot of new Dictionary functionalities that are not covered.

Reply:
> I took the liberty of merging the master but now it's even worse - oops.

Thanks …

Reply:
@TwoLaid okay the test coverage decreases now only by ~1.3%. The true coverage is higher, because a lot of templated method are in fact covered. We need to fix somehow, but maybe not in this pr.